std/assembly/date.ts

```diff
@@ -1,38 +1,249 @@
-import {
-  UTC as Date_UTC,
-  now as Date_now
-} from "./bindings/Date";
+import { E_VALUEOUTOFRANGE } from "util/error";
+import { now as Date_now } from "./bindings/Date";
 
 export class Date {
-
   @inline static UTC(
     year: i32,
     month: i32 = 0,
     day: i32 = 1,
     hour: i32 = 0,
     minute: i32 = 0,
     second: i32 = 0,
-    millisecond: i64 = 0
+    millisecond: i32 = 0
   ): i64 {
-    return <i64>Date_UTC(year, month, day, hour, minute, second, <f64>millisecond);
+    return epochMillis(year, month + 1, day, hour, minute, second, millisecond);
   }
 
   @inline static now(): i64 {
     return <i64>Date_now();
   }
 
-  private value: i64;
+  static fromString(dateTimeString: string): Date {
+    let hour: i32 = 0,
+        minute: i32 = 0,
+        second: i32 = 0,
+        millisecond: i32 = 0;
+    let dateString: string;
+
+    if (dateTimeString.includes("T")) {
+      // includes a time component
+      const parts = dateTimeString.split("T");
```

Comment:
There may be an opportunity here to make `fromString` basically allocation-free, by walking the string and using `strtol` to extract the values (currently doesn't support a limit, though). @MaxGraey Thoughts?

Reply:
yeah, instead `split` better use `while + indexOf` loop here

Reply:
Note that there would still be other avoidable allocations then when `substring`ing the parts of interest into `parseInt`. Hence I thought this may be a case for slightly extending `strtol` so it stops at a given limit :)

Reply:
I'll leave this one to you @MaxGraey  ;-)

Reply:
@ColinEberhardt ok